code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -753,3 +760,10 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 	targeted.Move(loc_b, get_dir(loc_b, loc_a))
 	new /obj/effect/temp_visual/xenomorph/afterimage(loc_a, targeted)
 	INVOKE_NEXT_TICK(src, PROC_REF(rewind))
+
+/datum/action/ability/activable/xeno/rewind/proc/handle_fire()
+	if(target_initial_on_fire == TRUE && target_initial_fire_stacks >= 0)
+		targeted.fire_stacks = target_initial_fire_stacks
+		targeted.IgniteMob()
+	else
+		targeted.ExtinguishMob()
```

Comment:
1) You can inline this (just copy paste the code where the proc is called) since its only used in one spot
2) You dont need to do `if(var == TRUE)` since in all but extreme cases `if(var)` does the same thing

Reply:
1. it ain't fancy
2. it for some reason doesn't work as i want it to with just `!target_initial_on_fire`